tools/ci_connector_ops/ci_connector_ops/pipelines/pipelines/metadata.py

```diff
@@ -212,3 +262,34 @@ async def run_metadata_upload_pipeline(
             pipeline_context.test_report = TestReport(pipeline_context, results)
 
         return pipeline_context.test_report.success
+
+
+async def run_metadata_orchestrator_deploy_pipeline(
+    is_local: bool,
+    git_branch: str,
+    git_revision: str,
+    gha_workflow_run_url: Optional[str],
+    pipeline_start_timestamp: Optional[int],
+    ci_context: Optional[str],
+) -> bool:
+    metadata_pipeline_context = PipelineContext(
+        pipeline_name="Metadata Service Orchestrator Unit Test Pipeline",
+        is_local=is_local,
+        git_branch=git_branch,
+        git_revision=git_revision,
+        gha_workflow_run_url=gha_workflow_run_url,
+        pipeline_start_timestamp=pipeline_start_timestamp,
+        ci_context=ci_context,
+    )
+
+    async with dagger.Connection(DAGGER_CONFIG) as dagger_client:
+        metadata_pipeline_context.dagger_client = dagger_client.pipeline(metadata_pipeline_context.pipeline_name)
+
+        async with metadata_pipeline_context:
+            steps = [
+                TestOrchestrator(context=metadata_pipeline_context),
+                DeployOrchestrator(context=metadata_pipeline_context)
+            ]
```

Comment:
I'm wondering if we shall make `DeployOrchestrator` depend on `TestOrchestrator` success explicitely or make it depend on the full orchestrator testing pipeline by calling `run_metadata_orchestrator_test_pipeline` and checking the success of the pipeline before running `DeployOrchestrator`.

I feel like the later is more maintainable: if  we add more tests or steps to `run_metadata_orchestrator_test_pipeline`  we won't have to update `run_metadata_orchestrator_deploy_pipeline`.

The current problem is that `run_metadata_orchestrator_deploy_pipeline` and `run_metadata_orchestrator_test_pipeline` will instantiate two differents dagger clients. Instantiating two different clients can lead to strange edge cases as two different client might have a different perceptions of the filesystem state if file changes between the two instantiation. 
I think it's not risky at the moment, but we might want to pass the dagger client directly to the `run_metadata_orchestrator_*` functions.

What do you think?

Reply:
I like this, but perhaps we do this during our larger refactor when we might tackle the idea of steps are pipelines.

Reply:
Sounds good to me